src/Enzo/enzo_EnzoMethodPmDeposit.cpp

```diff
@@ -252,53 +256,49 @@ void EnzoMethodPmDeposit::compute ( Block * block) throw()
 
       	  const int dp =  particle.stride(it,ia_x);
       	  const int dv =  particle.stride(it,ia_vx);
-
-	        for (int ip=0; ip<np; ip++) {
-
+	  
+	  for (int ip=0; ip<np; ip++) {
+	    
       	    double x = xa[ip*dp] + vxa[ip*dv]*dt;
       	    double y = ya[ip*dp] + vya[ip*dv]*dt;
 
       	    double tx = nx*(x - xm) / (xp - xm) - 0.5;
       	    double ty = ny*(y - ym) / (yp - ym) - 0.5;
-
+	    
       	    int ix0 = gx + floor(tx);
       	    int iy0 = gy + floor(ty);
-
+	    
       	    int ix1 = ix0 + 1;
       	    int iy1 = iy0 + 1;
-
+	    
       	    double x0 = 1.0 - (tx - floor(tx));
       	    double y0 = 1.0 - (ty - floor(ty));
-
+	    
       	    double x1 = 1.0 - x0;
       	    double y1 = 1.0 - y0;
-
-	          if ( dens < 0.0) {
-	            CkPrintf ("%s:%d ERROR: dens = %f\n", __FILE__,__LINE__,dens);
-	          }
-
-      	    de_p[ix0+mx*iy0] += pdens[ip]*dens*x0*y0;
-      	    de_p[ix1+mx*iy0] += pdens[ip]*dens*x1*y0;
-      	    de_p[ix0+mx*iy1] += pdens[ip]*dens*x0*y1;
-      	    de_p[ix1+mx*iy1] += pdens[ip]*dens*x1*y1;
-
+	    
+      	    de_p[ix0+mx*iy0] += pmass[ip*dm]*inv_vol*x0*y0;
+      	    de_p[ix1+mx*iy0] += pmass[ip*dm]*inv_vol*x1*y0;
+      	    de_p[ix0+mx*iy1] += pmass[ip*dm]*inv_vol*x0*y1;
+      	    de_p[ix1+mx*iy1] += pmass[ip*dm]*inv_vol*x1*y1;
+	    
       	    if ( de_p[ix0+mx*iy0] < 0.0) {
       	      CkPrintf ("%s:%d ERROR: de_p %d %d = %f\n",
-      	   	        __FILE__,__LINE__,ix0,iy0,dens);
+      	   	        __FILE__,__LINE__,ix0,iy0,inv_vol);
```

Comment:
Instead of `inv_vol` that's always positive, it might be more informative to report on the negative density (`de_p[ix0+mx*iy0]`) and/or mass (`pmass[ip*dm]`). Applies to all instances below, too.

From the previous version, I also see that this error check only happens in 2D. It should be consistently checked (or not) regardless of the rank. It doesn't need to be fixed in this PR, but I'll make an issue.

Reply:
Thanks for pointing this out, I think this is an artifact of some find-replaces I did.